### ISSUE TITLE
Получение значений тега из TLVDataHex

### ISSUE THREAD

**Issue by ConsperSerj** [shtrih-m/fr_drv_ng_examples#40]
Добрый день. Подскажите пожалуйста. 

Выполняю функцию проверки маркировки.
Пример кода: 
 Set_BarCode(params.markingCode.toString())
 Set_ItemStatus(params.plannedGoodStatus.code)
 Set_CheckItemMode(0)
 var tlvData = byteArrayOf()
Set_TLVDataHex("")
params.measureOfQuantity?.let {
    tlvData += getTLVDataHexForByte(2108, it.code.toLong())
}
params.quantity?.let { tlvData += getTLVDataHexForFVLN(1023, it)  }
params.fractionalCountOfMarkedPosition?.let { fractional ->
    Set_TagNumber(1291)
    Set_TagType(8)
    FNBeginSTLVTag()
    val parentId = Get_TagID()
    addTagVlnType(parentId, 1293, fractional.numerator.toString())
    addTagVlnType(parentId, 1294, fractional.denominator.toString())
    Set_TagNumber(parentId)
    Set_TagType(8)
    GetTagAsTLV()
    tlvData += Get_TLVData()
}
Set_TLVData(tlvData)
 FNCheckItemBarcode()
 
 Функция отрабатывает и на выходе получаю TLVDataHex. Подскажите пожалуйста возможно ли с TLVDataHex как то получить значения по тегу. Например я передавал тег 2108 в него значение 4. Как мне получить значение тега 2108 от ядра из TLVDataHex? Есть ли такая возможность?

**Comment by swex** [comment]
нет, такой возможности сейчас нет, мы умеем интерпретировать TLV, но только полученные из ККТ(метод `FNGetDocumentAsString`), снаружи нельзя передать. 

в общем-то можно добавить

**Comment by ConsperSerj** [comment]
Вот метод FNCheckItemBarcode заполняет TLVDataHex. Было бы очень хорошо чтобы например как и раньше я вызывал метод FNGetDocumentAsString и он мне возвращал это в строковом формате все данные ответа. Вы сможете что-то такое добавить?

**Comment by ConsperSerj** [comment]
Или может у вас есть алгоритм на java который позволит TLVHEX 
3C 08 01 00 00 FF 03 04 00 06 00 09 3D 0B 05 0A 00 0D 05 01 00 00 0E 05 01 00 00
Перегнать в map тег значение?

**Comment by swex** [comment]
в след. релиз добавим метод
```c++
    //!
    //! @brief Декодировать TLV в виде строки
    //!
    //! @par Используемые свойства:
    //!  - #TLVData - TLV для декодирования данные
    //!  - #ShowTagNumber - Показывать номер тэга.
    //!
    //! @par Модифицируемые свойства:
    //!  - #StringForPrinting - Cтрока с перечислением тегов ФН и их значений, формат зависит от DFE_DataPresentation
    //!
    //! Выдает TLV или в виде строк разделенных \n или в виде json.
    //! Вид представления задается методом #SetDeviceFunction функцией DFE_DataPresentation.
    //!
    //! @return Код ошибки
    //!
    int DecodeTLVData();
```

**Comment by ConsperSerj** [comment]
Подскажите пожалуйста а когда планируется следующий релиз?

**Comment by swex** [comment]
> Подскажите пожалуйста а когда планируется следующий релиз?

сегодня-завтра

**Comment by swex** [comment]
[1.6.0-5-g6b15123](https://github.com/shtrih-m/fr_drv_ng/releases/tag/1.6.0-5-g6b15123)
